sdl-core/src/main/scala/io/smartdatalake/definitions/Environment.scala

```diff
@@ -604,16 +604,24 @@ object Environment extends SmartDataLakeLogger {
     , "executionMode.checkpointLocation", "execution-mode.checkpoint-location")
   val runIdPartitionColumnName = "run_id"
 
-  // instantiate sdl plugin if configured
-  private[smartdatalake] def sdlPlugin: Option[SDLPlugin] = {
-    if (_sdlPlugin.isEmpty) {
-      _sdlPlugin = Some(EnvironmentUtil.getSdlParameter("pluginClassName")
-        .map(CustomCodeUtil.getClassInstanceByName[SDLPlugin]))
 
+  // instantiate sdl plugins if configured. The class names must be separated by a comma.
+  // initializes with envVars pluginClassName (deprecated) and pluginClassNames
+  private[smartdatalake] def sdlPlugins: Option[Seq[SDLPlugin]] = {
```

Comment:
```suggestion
  private[smartdatalake] def sdlPlugins: Seq[SDLPlugin] = {
```
Hier würde ich einfach eine Seq[SDLPlugin] zurückgeben. Dann brauchst du SmartDataLakeBuilder.scala nicht doppelte foreach.
In Zeile 614 solltest du dann _sdlPlugin.get lassen. Das ist an dieser Stelle ja dann immer definiert.

Reply:
removed the Option Type (see https://github.com/smart-data-lake/smart-data-lake/pull/969/commits/27f10e18549991333a969bfde71c6339a22c18ec) but sdlPlugin doesn't exist anymore, only sdlPlugins (which accepts two env variables).